### PR TITLE
feat/refactor: maze function changes to help with mapping

### PR DESCRIPTION
**Pull request by RosyMapleMoth**
#### Current State <span><a href="#details">( scroll down )</a></span>
- [x] Final

<h1 id="details">Details about this Pull Request</h1>

#### What's this Pull Request do?

Add a Node getter function to Maze, change the Neighbor Node function to a public function.
#### Any background context you want to provide?

While working with mapping it became apparent these changes would be very helpful/necessary  
#### Changes or additions made:
- [added]
  - Node\* getNode( Vector::Pos pos ); 
- [changed] 
  - Node\* getNeighborNode( Vector::Pos pos , direction dir );

### CODE REVIEW

File: Maze.cpp [hunk 1]
```diff
@@ -116,6 +116,11 @@ namespace Micromouse
 		return findPath( start->getPos() , end->getPos() );
 	}
 
+	// Retruns the Node at the given position.
+	Node* Maze::getNode( Vector::Pos pos )
+	{
+		return maze[ pos.x ][ pos.y ];
```
Comment (blhough) [inline Maze.cpp]: Careful x and y are private for Vector::Pos so you actually need to call functions to get x,y. 

pos.x() , pos.y()

**Comment by blhough** [comment]
Other than that one small thing i line commented on. This is a very nice PR and nice code!

I will merge once fixed. 
Any new commits to this branch are automatically added to the PR

**Comment by blhough** [comment]
what IDE or compiler are you using?
Make sure to compile the project before making a PR to ensure everything is working

**Comment by RosyMapleMoth** [comment]
I'm using visual studio 2015 but I hadn't attempted a compile because it was such a small change. my bad, and it won't happen again!

also I've just fixed the getNode

**Pull request merged**